watertap/core/watertap_flowsheet_block.py

```diff
@@ -0,0 +1,177 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+__author__ = "Alexander V. Dudchenko"
+
+from idaes.core.base.flowsheet_model import FlowsheetBlockData
+
+from idaes.core import (
+    declare_process_block_class,
+)
+from pyomo.common.config import ConfigValue
+
+from watertap.core.util.connections import (
+    PortContainer,
+    ConnectionContainer,
+)
+from watertap.core.util.report import (
+    build_report_table,
+)
+
+
+@declare_process_block_class("WaterTapFlowsheetBlock")
+class WaterTapFlowsheetBlockData(FlowsheetBlockData):
+    CONFIG = FlowsheetBlockData.CONFIG()
+    CONFIG.declare(
+        "default_costing_package",
+        ConfigValue(
+            default=None,
+            description="defines default costing package",
+            doc="""
+                defines default costing package
+            """,
+        ),
+    )
+    CONFIG.declare(
+        "default_costing_package_kwargs",
+        ConfigValue(
+            default={},
+            description="kwargs to pass into the Costing unit block",
+            doc="""
+                kwargs to pass into the Costing unit block,
+            """,
+        ),
+    )
+
+    def build(self):
+        self.outlet_connections = []
+        super().build()
+
+    def fix_and_scale(self):
+        self.set_fixed_operation()
+        self.scale_before_initialization()
+
+    def initialize_unit(self, **kwargs):
```

Comment:
Should have the methods that are required raise a `NotImplementedError` in this base